rubyparser.h

```diff
@@ -34,6 +34,14 @@
 #endif
 #endif
 
+#if defined(__GNUC__)
+#   define RUBYPARSER_ATTRIBUTE_FORMAT(string_index, argument_index) __attribute__((format(printf, string_index, argument_index)))
+#elif defined(__clang__)
+#   define RUBYPARSER_ATTRIBUTE_FORMAT(string_index, argument_index) __attribute__((__format__(__printf__, string_index, argument_index)))
+#else
+#   define RUBYPARSER_ATTRIBUTE_FORMAT(string_index, argument_index)
+#endif
```

Comment:
`__MINGW_PRINTF_FORMAT` would be more preferable if defined.

Reply:
@nobu 
Thank you!
I added https://github.com/ruby/ruby/pull/10231/files#diff-daa4638f1a96635ab424e04973b51e448c0caee56f8cdfed8fe2e96e25f1e8b6R38-R40